### PR TITLE
fix: test cases to have empty objects and not nulls

### PR DESCRIPTION
**Pull request by derberg**
**Description**

- test cases to have empty objects and not nulls

**Related issue(s)**
Fixes https://github.com/asyncapi/parser-js/issues/144
Fixes https://github.com/asyncapi/tck/issues/25

### CODE REVIEW

**Comment by derberg** [comment]
2 test cases from this PR are a `?` to me:
- `tests/asyncapi-2.0/Message Trait Object/invalid-missing-content-type.yaml`
- `tests/asyncapi-2.0/Message Object/invalid-missing-contentType.yaml`

`contentType` and `defaultContentType` fields are not required by the spec, so in my opinion these test cases should be removed, probably in another PR for clarity? as I think it would mean we also remove `tests/asyncapi-2.0/Message Object/valid-using-defaultContentType.yaml ` and ` tests/asyncapi-2.0/Message Trait Object/valid-using-defaultContentType.yaml`

**Comment by postatum** [comment]
> 2 test cases from this PR are a `?` to me:
> 
>     * `tests/asyncapi-2.0/Message Trait Object/invalid-missing-content-type.yaml`
> 
>     * `tests/asyncapi-2.0/Message Object/invalid-missing-contentType.yaml`
> 
> 
> `contentType` and `defaultContentType` fields are not required by the spec, so in my opinion these test cases should be removed, probably in another PR for clarity? as I think it would mean we also remove `tests/asyncapi-2.0/Message Object/valid-using-defaultContentType.yaml ` and ` tests/asyncapi-2.0/Message Trait Object/valid-using-defaultContentType.yaml`

I've added these tests because, for example, [this section](https://github.com/asyncapi/asyncapi/blob/master/versions/2.0.0/asyncapi.md#messageTraitObject) says:

> The content type to use when encoding/decoding a message's payload. The value MUST be a specific media type (e.g. application/json). When omitted, the value MUST be the one specified on the defaultContentType field.

Thus I assumed that either `contentType` or `defaultContentType` must be specified.

Feel free to open an issue so we can discuss this further.

File: tests/asyncapi-2.0/Message Object/valid-empty-object.yaml [hunk 1]
```diff
@@ -1,10 +1,12 @@
 asyncapi: 2.0.0
 
+defaultContentType: application/json
```
Comment (fmvilas) [inline tests/asyncapi-2.0/Message Object/valid-empty-object.yaml:3]: Why is this important in this test case?

**Comment by fmvilas** [comment]
I think you found a gap in the spec, @postatum. There should be a default value for `defaultContentType` or otherwise make it required (which I don't like). Can you create an issue on asyncapi/asyncapi repo?

**Comment by postatum** [comment]
> I think you found a gap in the spec, @postatum. There should be a default value for `defaultContentType` or otherwise make it required (which I don't like). Can you create an issue on asyncapi/asyncapi repo?

Done https://github.com/asyncapi/asyncapi/issues/421

**Comment by fmvilas** [comment]
Thanks, mate!

**Comment by derberg** [comment]
created this issue to followup on the discussion https://github.com/asyncapi/tck/issues/27

**Pull request merged**